packages/template-retail-react-app/app/components/recommended-products/index.jsx

```diff
@@ -104,10 +110,20 @@ const RecommendedProducts = ({zone, recommender, products, title, shouldFetch, .
     // TODO: DRY this handler when intl provider is available globally
     const addItemToWishlist = async (product) => {
         try {
-            await wishlist.createListItem({
-                id: product.productId,
-                quantity: 1
+            await createCustomerProductListItem.mutateAsync({
+                parameters: {
+                    listId: wishlist.id,
```

Comment:
`wishlist` is possibly be undefined.
```suggestion
                    listId: wishlist?.id,
```